gopherage/pkg/cov/junit/calculation/coveragelist.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package calculation
+
+import (
+	"path"
+	"sort"
+	"strings"
+)
+
+// CoverageList is a collection and summary over multiple file Coverage objects
+type CoverageList struct {
+	*coverage
+	Group             []coverage
+	CoverageThreshold int
+}
+
+// CovList constructs new (file) Group Coverage
+func newCoverageList(name string, covThresholdInt int) *CoverageList {
+	return &CoverageList{
+		coverage:          &coverage{Name: name},
+		Group:             []coverage{},
+		CoverageThreshold: covThresholdInt,
+	}
+}
+
+func (covList *CoverageList) ratio() float32 {
+	covList.Summarize()
+	return covList.coverage.ratio()
+}
+
+// Summarize summarizes all items in the Group and stores the result
+func (covList *CoverageList) Summarize() {
```

Comment:
Calling this more than once will result in `nCoveredStmts` and `nAllStmts` being wrong. They should probably be zeroed out first.

I also think it is a confusing design that `nCoveredStmts` and `nAllStmts` are invalid until this is called (and become invalid again when this is called a second time). I would prefer a design where either:

a) these were calculated on the fly when they are needed (this has worked well for me), or if that is too inefficient,
b) these are _conceptually_ calculated on the fly as needed, but cached (with appropriate invalidation)

Reply:
Added summaryCompleted property to be checked against to see if the summary has already been computed. If so, then use the cached result. Otherwise do the calculation and set the property to true.